lib/matobj.gi

```diff
@@ -1380,6 +1380,38 @@ InstallMethod( InverseSameMutability,
     fi;
     end );
 
+InstallMethod( IsZero,
+    [ IsRowListMatrix and IsMatrixObj ],
+    function( mat )
+    local ncols, row;
+
+    ncols:= NrCols( mat );
+    for row in mat do
+      if PositionNonZero( row ) <= ncols then
+        return false;
+      fi;
+    od;
+    return true;
+    end );
+
+InstallMethod( IsOne,
+    [ IsRowListMatrix and IsMatrixObj ],
+    function( mat )
+    local ncols, i, row;
+
+    ncols:= NrCols( mat );
+    for i in [1 .. NrRows( mat )] do
+      row := mat[i];
+      if PositionNonZero( row ) <> i or not IsOne( row[i] ) then
+        return false;
+      fi;
+      if PositionNonZero( row, i ) <= ncols then
```

Comment:
Think you might have copy pasted lines 1408 to 1410 from `IsZero` here, this can't be right.

Reply:
Sorry my mistake.